### DISCUSSION TITLE
Wie ruft man aha.sh auf?

### DISCUSSION THREAD

**Discussion by leo-022** [Freetz-NG/freetz-ng#360] [Allgemein]
In #350 schrieb @fda77 _Wirf einfach die kompletten AVM-Einstellungen raus. Die sind viel zu unflexibel. Das Mausgeschubse nervt auch. Mit "cron" und "aha.sh" kannst du unendlich viele Schaltzeitpunkte einstellen._

Ich habe nun etwas mit aha.sh probiert, Ergenis war, dass ich alle meine SmartHome-Geräte gelöscht habe. Leider fine ich zu aha.sh nur
```
Usage: /usr/bin/aha.sh <alias|fancy|small|battery|modus|gradc|translate <ain|name>|docmd <device> <command|-1|0|1|16-56\|8,0-28,0> >
```
Kann jemand ein paar Beispiele zum Aufruf von aha.sh geben? Ich vermute, @fda77, du hast dazu Einges in deiner crontab stehen.

**Comment by fda77** [comment]
Täglich um 8 NAME schalten. Ob das mit Leerzeichen im Namen funktioniert hab ich nicht getestet
In Cron:
```
0 8    * * *  /usr/bin/aha.sh docmd NAME   32
oder
0 8    * * *  /usr/bin/aha.sh docmd NAME   16,0
```

Mit `docmd` kann man auch Steckdosen schalten, dann -1/0/1 oder toggle/off/on als Parameter
Ansonsten ist für Benutzer `battery` noch interessant, das könnte man mit einem Script auswerten

Solang niemand eine readme/wiki schreibt ist der Source die Doku!

**Comment by leo-022** [comment]
Ich habe da mal eine kurze readme geschrieben.

Die Steuerung der SmartHome-Geräte geschieht mit aha.sh. Man kann z. B. per cron mit aha.sh entsprechende Schaltzeitpunkte einstellen.
Derzeit unterstützt aha.sh nur Heizkörperregler wie Comet DECT, FRITZ!DECT 300 oder FRITZ!DECT 301 und ein/aus-Aktoren wie die Steckdosen FRITZ!DECT 200 oder FRITZ!DECT 210.

Parameter für den Aufruf von aha.sh

**a** oder **alias**
sha-alias (Liste aller Smarthome-Geräte) aktualisieren (dasselbe wie „Smarthome aktualisieren“ im Webinterface)

**f** oder **fancy**
Ausgabe der grundlegenden Informationen aller SmartHome-Geräte
 ```
            manufacturer = AVM
             productname = Comet DECT
               fwversion = 03.68
         functionbitmask = 320
                           ---- ---- -1-1 ----- -
              identifier = 11111 0112314
                      id = 20
                    name = Heizung A
                 present = 1
                    lock = 0
              devicelock = 0
                 celsius = 270
                  offset = 0
                    tist = 54
                   tsoll = 253
                  absenk = 32
                 komfort = 44
         windowopenactiv = 0
                 battery = 80
              batterylow = 0


...


                           HGFE DCBA 9876 54321 0 
                           |||| |||| |||| ||||  | 
                           |||| |||| |||| ||||  +- Bit  0: HANFUN Gerät
                           |||| |||| |||| ||||    
                           |||| |||| |||| |||+---- Bit  2: Lampe
                           |||| |||| |||| ||+----- Bit  3: ?Action
                           |||| |||| |||| |+------ Bit  4: Alarmsensor
                           |||| |||| |||| +------- Bit  5: Taster
                           |||| |||| ||||         
                           |||| |||| |||+--------- Bit  6: Heizkörperregler
                           |||| |||| ||+---------- Bit  7: Energiemessgerät
                           |||| |||| |+----------- Bit  8: Temperatursensor
                           |||| |||| +------------ Bit  9: Schaltsteckdose
                           |||| ||||              
                           |||| |||+-------------- Bit 10: DECT-Repeater
                           |||| ||+--------------- Bit 11: Mikrofon
                           |||| |+---------------- Bit 12: ?Bundle
                           |||| +----------------- Bit 13: HANFUN Unit
                           ||||                   
                           |||+------------------- Bit 14: ?Template
                           ||+-------------------- Bit 15: Schaltbar
                           |+--------------------- Bit 16: Dimmbar
                           +---------------------- Bit 17: Farbtemperatur
```
**s** oder **small**
Ausgabe von ain, Name, aktuelle Leistung, aktuelle Spannung, absoluter Verbrauch seit Zurücksetzen der Energiestatistik, Temperatur, Temperaturabweichung (Offset), Current, Factor
```
111110112314|Heizung A||||270|0||
111110222314|Heizung B||||275|0||
111300022222|Steckdose|45910|230676|31986|280|0|2820|706
```

**b** oder **battery**
Ausgabe von ain, Name, Batterieladezustand in Prozent, Batterieladezustand niedrig (=1)
```
111110112314|Heizung A|80|0
111110222314|Heizung B|80|0
```

**m** oder **modus**
Ausgabe von ain, Name, Schaltzustand, Lock
```
111110112314|Heizung A||0
111110222314|Heizung B||0
111300022222|Steckdose|1|0
grp7D7D7D-1C991C990|Heizungen||0
```

**g** oder **gradc**
Ausgabe von ain, Name, Schaltzustand, Lock, Isttemperatur in 1/10 °C, Solltemperatur in internen Temperatureinheiten (8°C=16, 28°C=56, aus=253, ein=254)
```
111110112314|Heizung A||0|270|32
111110222314|Heizung B||0|270|32
111300022222|Steckdose|1|0|275|
grp7D7D7D-1C991C990|Heizungen||0||253
```

**t** oder **translate**
ain <-> Name zuordnen oder umgekehrt
```
aha.sh t 111300022222
Steckdose
```

```
aha.sh t Steckdose
111300022222
```

**d** oder **docmd**
SmartHome-Geräte schalten

Zusätzlich erforderliche Parameter: Device Command oder Wert
Device kann als Name oder ain angegeben werden.

Command kann für ein/aus-Aktoren wie folgt lauten:
- off (aus)
- on (ein)
- toggle (umschalten)

Bereich für Wert bei ein/aus-Aktoren:
- 0 (aus)
- 1 (ein)
- -1 (umschalten)

Bereich für Wert bei Heizkörperreglern:
- 16…56 (interne Temperatureinheiten)
- 8.0…28.0 (°C)
- 253 (aus)
- 254 (ein)

`aha.sh d Steckdose off`
Steckdose aus

`aha.sh d Steckdose 1`
Steckdose ein

`aha.sh d Steckdose toggle`
Umschalten der Steckdose ein/aus

`aha.sh d "Heizung A" 17`
Heizung A auf 16 internen Einheiten schalten (8,5°C)

`aha.sh d 111110112314 8,5`
Heizung mit ain 111110112314 auf 8,5°C schalten

`aha.sh d Heizungen 253`
Gruppe Heizungen ausschalten

**Comment by fda77** [comment]
Danke, ich hab es hier reingepackt: https://freetz-ng.github.io/freetz-ng/make/rrdstats.html#script-ahash
Kleine Änderungen wie 546e
Für Updates unter docs/ bei der entsprechenden Datei den edit-Stift usw
Dect 400 wird auch unterstützt, ausser Batterieanzeige kann der aber nichts
